pkg/resources/streamlit.go

```diff
@@ -0,0 +1,403 @@
+		return nil, err
+	}
+	if err = d.Set("title", streamlit.Title); err != nil {
+		return nil, err
+	}
+	if err = d.Set("comment", streamlit.Comment); err != nil {
+		return nil, err
+	}
+	return []*schema.ResourceData{d}, nil
+}
+
+func CreateContextStreamlit(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*provider.Context).Client
+	databaseName := d.Get("database").(string)
+	schemaName := d.Get("schema").(string)
+	name := d.Get("name").(string)
+	id := sdk.NewSchemaObjectIdentifier(databaseName, schemaName, name)
+	stageId := sdk.NewSchemaObjectIdentifierFromFullyQualifiedName(d.Get("stage").(string))
+	rootLocation := fmt.Sprintf("@%s", stageId.FullyQualifiedName())
+	if v, ok := d.GetOk("directory_location"); ok {
+		rootLocation = path.Join(rootLocation, v.(string))
+	}
+	req := sdk.NewCreateStreamlitRequest(id, rootLocation, d.Get("main_file").(string))
+
+	if v, ok := d.GetOk("query_warehouse"); ok {
+		req.WithQueryWarehouse(sdk.NewAccountObjectIdentifier(v.(string)))
+	}
+	if v, ok := d.GetOk("title"); ok {
+		req.WithTitle(v.(string))
+	}
+	if v, ok := d.GetOk("comment"); ok {
+		req.WithComment(v.(string))
+	}
+	if v, ok := d.GetOk("external_access_integrations"); ok {
+		raw := expandStringList(v.(*schema.Set).List())
+		integrations := make([]sdk.AccountObjectIdentifier, len(raw))
+		for i, v := range raw {
+			integrations[i] = sdk.NewAccountObjectIdentifier(v)
+		}
+		req.WithExternalAccessIntegrations(sdk.ExternalAccessIntegrationsRequest{
+			ExternalAccessIntegrations: integrations,
+		})
+	}
+	if err := client.Streamlits.Create(ctx, req); err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(helpers.EncodeSnowflakeID(id))
+
+	return ReadContextStreamlit(false)(ctx, d, meta)
+}
+
+func ReadContextStreamlit(withExternalChangesMarking bool) schema.ReadContextFunc {
+	return func(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+		client := meta.(*provider.Context).Client
+		id := helpers.DecodeSnowflakeID(d.Id()).(sdk.SchemaObjectIdentifier)
+
+		streamlit, err := client.Streamlits.ShowByID(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to query streamlit. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("Streamlit name: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.FromErr(err)
+		}
+
+		streamlitDetails, err := client.Streamlits.Describe(ctx, id)
+		if err != nil {
+			if errors.Is(err, sdk.ErrObjectNotFound) {
+				d.SetId("")
+				return diag.Diagnostics{
+					diag.Diagnostic{
+						Severity: diag.Warning,
+						Summary:  "Failed to query streamlit properties. Marking the resource as removed.",
+						Detail:   fmt.Sprintf("Streamlit name: %s, Err: %s", id.FullyQualifiedName(), err),
+					},
+				}
+			}
+			return diag.FromErr(err)
+		}
```

Comment:
where is the part with withExternalChangesMarking?
 - are there no underlying snowflake defaults we have to reset to?
 - in such case, do we need this param in the read func?

Reply:
That's correct - removed.